src/test/java/com/google/sps/servlets/PotentialMatchesServletTest.java

```diff
@@ -0,0 +1,233 @@
+  private static final String NO_POTENTIAL_MATCH_RESULT = "NO_POTENTIAL_MATCHES";
+
+  private static final String USER_ENTITY = "User";
+  private static final String USER_FRIENDS_LIST_PROPERTY = "friends-list";
+  private static final String USER_ID_PROPERTY = "id";
+  private static final String USER_NAME_PROPERTY = "name";
+  private static final String USER_EMAIL_PROPERTY = "email";
+  private static final String USER_BIO_PROPERTY = "bio";
+
+  private static final String TEST_USER_1_NAME = "Daenerys";
+  private static final String TEST_USER_1_ID = "5555";
+  private static final String TEST_USER_1_EMAIL = "targaryen@gmail.com";
+  private static final String TEST_USER_1_BIO = "dragons <3";
+
+  private static final String TEST_USER_2_NAME = "Eliza";
+  private static final String TEST_USER_2_ID = "1776";
+  private static final String TEST_USER_2_EMAIL = "elizaHam@gmail.com";
+  private static final String TEST_USER_2_BIO = "hamilFam";
+
+  private static final String TEST_USER_3_NAME = "Rory";
+  private static final String TEST_USER_3_ID = "1234";
+  private static final String TEST_USER_3_EMAIL = "gilmore@gmail.com";
+  private static final String TEST_USER_3_BIO = "Stars Hallow :)";
+
+  private static final String TEST_USER_4_NAME = "Steve";
+  private static final String TEST_USER_4_ID = "1964";
+  private static final String TEST_USER_4_EMAIL = "capAmerica@gmail.com";
+  private static final String TEST_USER_4_BIO = "Avengers, assemble";
+
+  private final LocalServiceTestHelper helper =
+    new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Mock
+  private HttpServletRequest mockRequest;
+ 
+  @Mock
+  private HttpServletResponse mockResponse;
+
+  private PotentialMatchesServlet servletUnderTest;
+  private DatastoreService datastore;
+ 
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+    servletUnderTest = new PotentialMatchesServlet();
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  /**
+  * Tests scenario where there is only one user in datastore
+  *
+  * <p>Should result in no next potential match found
+  */
+  @Test
+  public void oneUserNoPotentialMatches() throws Exception {
+    when(mockRequest.getParameter("userid")).thenReturn(TEST_USER_1_ID);
+
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    when(mockResponse.getWriter()).thenReturn(writer);
+
+    addTestUserEntityToDatastore(datastore, TEST_USER_1_ID, TEST_USER_1_NAME,
+      TEST_USER_1_EMAIL, TEST_USER_1_BIO, new String[]{});
+    
+    servletUnderTest.doGetWrapper(datastore, mockRequest, mockResponse);
+    writer.flush();
+
+    String expectedOutput = NO_POTENTIAL_MATCH_RESULT;
+    String actualOutput = stringWriter.toString();
+
+    assertThat(actualOutput).isEqualTo(expectedOutput);
```

Comment:
Probably not necessary to store these values in a String.

Reply:
I put these in separate String variables just for readability.

Reply:
I agree with having the `actualOutput` (`stringWriter.toSring()` is unclear), but the expected can be inlined. `isEqualTo` implies it's the expected.

Reply:
Got it!